src/effects/presets/effectpreset.cpp

```diff
@@ -99,5 +99,22 @@ const QDomElement EffectPreset::toXml(QDomDocument* doc) const {
     return effectElement;
 }
 
-EffectPreset::~EffectPreset() {
+void EffectPreset::updateParametersFrom(const EffectPreset& other) {
+    DEBUG_ASSERT(backendType() == other.backendType());
+
+    // technically algorithmically inefficient solution O(n²). May be
+    // optimizable by sorting first, gains depend on parameter count
```

Comment:
Can you clarify? Do you mean that we change `m_effectParameterPresets` to be a QHash instead of a `QList`? That won't work because we need to preserve the ordering of `m_effectParameterPresets`. What would work though is if we just built a temporary QHash on each call. What do you think?